WordPress/Classes/ViewRelated/Post/Revisions/RevisionsTableViewController.swift

```diff
@@ -154,16 +154,22 @@ private extension RevisionsTableViewController {
 
         SVProgressHUD.show(withStatus: NSLocalizedString("Loading...", comment: "Text displayed in HUD while a revision post is loading."))
 
-        let service = PostService(managedObjectContext: ContextManager.sharedInstance().mainContext)
-        service.getPostWithID(revision.revisionId, for: blog, success: { post in
-            SVProgressHUD.dismiss()
-            WPAnalytics.track(.postRevisionsRevisionLoaded)
-            self.onRevisionLoaded(post)
-            self.navigationController?.popViewController(animated: true)
-        }, failure: { error in
-            DDLogError("Error loading revision: \(error.localizedDescription)")
-            SVProgressHUD.showDismissibleError(withStatus: NSLocalizedString("Error occurred\nduring loading", comment: "Text displayed in HUD while a post revision is being loaded."))
-        })
+        let coreDataStack = ContextManager.shared
+        let postRepository = PostRepository(coreDataStack: coreDataStack)
+        Task { @MainActor in
+            do {
+                let postID = try await postRepository.getPost(withID: revision.revisionId, from: .init(saved: blog))
+                let post = try coreDataStack.mainContext.existingObject(with: postID)
```

Comment:
This two-step approach to get the post is repeated in https://github.com/wordpress-mobile/WordPress-iOS/pull/21439 and https://github.com/wordpress-mobile/WordPress-iOS/pull/21442.

Have you considered DRYing it? (I understand that might not have been possible because the PRs were parallel, which is very handy from a review point of view. Just a question for a next step).

Or, do we want to make the `TaggedManagedObjectID<T>` usage and the ability to pick the context to use explicit?

Reply:
In my ideal world, we don't pass Core Data model instances among functions. That means we should not need to call `existingObject(with:)` here. But we have to do that right now because that's how it's done at the moment: `PostService.getPost` returns a `Post` instance.

If we want to DRY this query, we'll need to move this query into the repository type, which is potentially error prone. Because it's unsafe to pass Core Data model to Swift's structured concurrency: [`NSManagedObject` is not `Sendable`](https://developer.apple.com/documentation/coredata/nsmanagedobject#conforms-to). That's why the PostRepository's API deals with [`NSManagedObjectID` which is `Sendable`](https://developer.apple.com/documentation/coredata/nsmanagedobjectid#conforms-to).


> do we want to make the TaggedManagedObjectID<T> usage and the ability to pick the context to use explicit?

Yes. It's safe to use mainContext here because the task is assigned to the `MainActor`.

Reply:
Gotcha. This is yet another bit of Core Data knowledge that I wish was more accessible to the casual user 😞